### PR TITLE
x86_cpu_model:update cpu model category

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 2632

includes a small fix and adds some versioned CPU models

### CODE REVIEW

**Comment by nanliu-r** [comment]
@PaulYuuu Hi, please help review this patch, thanks!

**Comment by nanliu-r** [comment]
Test passed:
1.  qemu-kvm-9.0.0 on Intel Sapphire Rapids HW for RHEL9
2.  qemu-kvm-6.2.0 on AMD Zen3 HW
3.  qemu-kvm-6.2.0 on Intel Cooper lake HW
4.  qemu-kvm-9.0.0 on AMD Zen3 HW for CentOS 10
5.  qemu-kvm-9.0.0 on AMD Zen4 HW for CentOS 10

File: qemu/tests/cfg/x86_cpu_model.cfg [hunk 1]
```diff
@@ -61,29 +61,44 @@
...
             model_pattern = "AMD EPYC-Rome-v3 Processor%s"
             RHEL.6, RHEL.7:
                 flags = ""
+        - EPYC-Rome-v4:
+            only HostCpuVendor.amd
+            required_qemu = [8.0.0,)
+            flags = "movbe rdrand rdtscp fxsr_opt cr8_legacy osvw fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt sha_ni xsaveopt xsavec xgetbv1 arat f16c xsaveerptr clzero rdpid perfctr_core ibpb wbnoinvd stibp clwb umip"
+            model_pattern = "AMD EPYC-Rome-v4 Processor%s"
+            RHEL.6, RHEL.7:
+                flags = "
```
Comment (PaulYuuu) [inline qemu/tests/cfg/x86_cpu_model.cfg:101]: typo?

**Review by PaulYuuu — COMMENTED** [review]
BTW, I would suggest using different commits to fix different issues.

File: qemu/tests/cfg/x86_cpu_model.cfg (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/cfg/x86_cpu_model.cfg:101]: > typo?

No, we support AMD Zen3 since RHEL 8, this is kind of negative test.

**Comment by nanliu-r** [comment]
> BTW, I would suggest using different commits to fix different issues.

Will this be a conflict since the update and small fix are in the same file?

File: qemu/tests/cfg/x86_cpu_model.cfg (hunk 1 above)
Comment (PaulYuuu) [reply qemu/tests/cfg/x86_cpu_model.cfg:101]: I mean the single `"`

Comment (nanliu-r) [reply qemu/tests/cfg/x86_cpu_model.cfg]: Correct it, thanks!

**Comment by nanliu-r** [comment]
Test pass with rhel vm and windows vm on latest centos 10.

@PaulYuuu please help review this , thanks.

**Comment by PaulYuuu** [comment]
Still, the PR fixes more than one issue, so I suggest split with 2 or more commits.

**Review by PaulYuuu — APPROVED** [review]

**Comment by nanliu-r** [comment]
regression test pass.

**Pull request merged**